Content.Shared/Interaction/SmartEquipSystem.cs

```diff
@@ -157,7 +158,8 @@ private void HandleSmartEquip(ICommonSession? session, string equipmentSlot)
         }
 
         // case 3 (itemslot item):
-        if (TryComp<ItemSlotsComponent>(slotItem, out var slots))
+        if (TryComp<ItemSlotsComponent>(slotItem, out var slots) &&
+            !HasComp<GunComponent>(slotItem))
```

Comment:
Guns shouldn't be hardcoded into smartequip. It should be adjusted to be a global behavior pulling the item itself if you want it.